container/go/pkg/compat/index.go

```diff
@@ -0,0 +1,198 @@
+	return types.OCIImageIndex, nil
+}
+
+// Digest returns the sha256 hash of this index's manifest.json metadata, an entrypoint for the config and layers.
+func (i *intermediateLayout) Digest() (v1.Hash, error) {
+	// We expect a file named digest that stores the manifest's hash formatted as sha256:{Hash} in this directory.
+	digest, err := ioutil.ReadFile(i.path.path(digestFile))
+	if err != nil {
+		fmt.Errorf("Failed to locate SHA256 digest file for image manifest: %v", err)
+	}
+
+	return v1.NewHash(string(digest))
+}
+
+// IndexManifest returns this image index's manifest object, simulates index.json in OCI layout.
+func (i *intermediateLayout) IndexManifest() (*v1.IndexManifest, error) {
+	// Parse raw manifest into a manifest struct.
+	manifest, err := partial.Manifest(i)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse raw manifest, please check if a correctly formatted manifest.json exists %v", err)
+	}
+	manifestDigest, err := i.Digest()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse image manifest hash, please check if a digest file exists in the directory and it is formatted as {Algorithm}:{Hash} %v", err)
+	}
+
+	// We are missing index.json in this intermediate format.
+	// Since index.json is represented in IndexManifest structure, we will populate this struct with parsed info.
+	index := v1.IndexManifest{
+		SchemaVersion: manifest.SchemaVersion,
+		Manifests: []v1.Descriptor{
+			v1.Descriptor{
+				MediaType: manifest.MediaType,
+				Size:      int64(len(i.rawManifest)),
+				Digest:    manifestDigest,
+			},
+		},
+	}
+
+	return &index, nil
+}
+
+// RawManifest returns the serialized bytes of manifest.json metadata.
+func (i *intermediateLayout) RawManifest() ([]byte, error) {
+	i.manifestLock.Lock()
+	defer i.manifestLock.Unlock()
+
+	if i.rawManifest == nil {
+		rawManifest, err := ioutil.ReadFile(i.path.path(manifestFile))
+		if err != nil {
+			return nil, err
+		}
+		i.rawManifest = rawManifest
+	}
+
+	return i.rawManifest, nil
+}
+
+// Image returns a v1.Image that this ImageIndex references.
+func (i *intermediateLayout) Image(h v1.Hash) (v1.Image, error) {
+	// Iterate through the list of manifests and get the manifest descriptor with digest h.
+	desc, err := i.findDescriptor(h)
+	if err != nil {
+		return nil, err
+	}
+
+	if !isExpectedMediaType(desc.MediaType, types.OCIManifestSchema1, types.DockerManifestSchema2) {
+		return nil, fmt.Errorf("unexpected media type for %v: %s", h, desc.MediaType)
+	}
+
+	img := &layoutImage{
+		path: i.path,
+		desc: *desc,
+	}
+
+	return partial.CompressedToImage(img)
+}
+
+// findDescriptor looks for the manifest with digest h in our "index.json" struct and returns its descriptor.
+func (i *intermediateLayout) findDescriptor(h v1.Hash) (*v1.Descriptor, error) {
+	im, err := i.IndexManifest()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, desc := range im.Manifests {
+		if desc.Digest == h {
+			return &desc, nil
```

Comment:
will we always only have one desc in this list and that value is the same as the one in `digest` file?